test/runtests.jl

```diff
@@ -222,29 +251,38 @@ end
     @testset "version mismatch" begin
         preserve_build() do
             # Mismatch in written file
+            let ROOTENV=joinpath(condadir, "3"), CONDA_EXE=default_conda_exe(ROOTENV)
```

Comment:
If you are using strange indentation to keep the diff minimal, I suggest `local` (it applies to other places)

```suggestion
            local ROOTENV=joinpath(condadir, "3")
            local CONDA_EXE=default_conda_exe(ROOTENV)
```